Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py

```diff
@@ -69,85 +172,215 @@ def highjack_transition_rando():
+        transitions_map[counterpart_original] = counterpart_redirect
+        origins_bucket.remove(counterpart_original)
+        redirections_bucket.remove(counterpart_redirect)
+
+
+def get_random_redirection(original: Transition, all_redirections: Iterable[Transition]):
+    possible_redirections = [
+        redirect for redirect in all_redirections
+        if original.from_ != redirect.to  # Prevent looping on itself
+    ]
+    if len(possible_redirections) > 0:
+        return random.choice(possible_redirections)
+    return None
 
 
 def set_transitions_map():
-    _possible_transitions_bucket = list(starmap(Transition, ALL_POSSIBLE_TRANSITIONS))
-    """A temporary container of transitions to pick from until it is empty."""
     transitions_map.clear()
     if not CONFIGS.SKIP_JAGUAR:
         starting_default = TRANSITION_INFOS_DICT[starting_area].default_entrance
         tutorial_original = Transition(from_=LevelCRC.JAGUAR, to=LevelCRC.PLANE_CUTSCENE)
         tutorial_redirect = Transition(from_=starting_default, to=starting_area)
         transitions_map[tutorial_original] = tutorial_redirect
 
-    for area in TRANSITION_INFOS_DICT.values():
-        for to_og in (exit_.area_id for exit_ in area.exits):
-            original = Transition(from_=area.area_id, to=to_og)
-            redirect = get_random_redirection(original, _possible_transitions_bucket)
-            if redirect is None or original in transitions_map:
-                # Don't override something set in a previous iteration,
-                # like from linked two-way entrances.
+    _possible_redirections_bucket = list(starmap(Transition, ALL_POSSIBLE_TRANSITIONS))
+    one_way_list = [
+        Transition(from_=LevelCRC.WHITE_VALLEY, to=LevelCRC.MOUNTAIN_SLED_RUN),
+        Transition(from_=LevelCRC.MOUNTAIN_SLED_RUN, to=LevelCRC.APU_ILLAPU_SHRINE),
+        Transition(from_=LevelCRC.APU_ILLAPU_SHRINE, to=LevelCRC.WHITE_VALLEY),
+        Transition(from_=LevelCRC.CAVERN_LAKE, to=LevelCRC.JUNGLE_CANYON),
+    ]
+    if CONFIGS.LINKED_TRANSITIONS:
+        # Ground rules:
+        # 1. you can't make a transition from a level to itself
+        # 2. any 2 levels may have a maximum of 1 connection between them (as long as it's 2-way)
+
+        _possible_origins_bucket = list(starmap(Transition, ALL_POSSIBLE_TRANSITIONS))
+
+        level_list: list[Area] = []
+        for area in TRANSITION_INFOS_DICT.values():
+            area.con_left = len(area.exits)
+            if area.con_left > 0:
+                level_list.append(area)
+        random.shuffle(level_list)
+        level_list.sort(key=lambda a: a.con_left, reverse=True)
+
+        link_list = [link_two_levels(level_list[0], level_list[1])]
+        global total_con_left
+        total_con_left = level_list[0].con_left + level_list[1].con_left
+
+        index = 2
+        while index < len(level_list):
+            choice = random.choice([Choice.CONNECT, Choice.INBETWEEN])
```

Comment:
The `Enum` class is iterable, so you can do stuff like this to avoid listing every value:
```suggestion
            choice = random.choice(tuple(Choice.CONNECT))
```

Reply:
Changed.